promql/functions.go

```diff
@@ -1010,6 +1097,12 @@ var functions = map[string]*Function{
 		ReturnType: model.ValVector,
 		Call:       funcMinOverTime,
 	},
+	"month_of_year": {
```

Comment:
This is a very far fetched example without any real use in practice. I'm against adding support for such exotic date functions to the query language. The definition of hour/day/month/year is very clear and established, there is no need to be overly explicit. Usability has to be kept in mind as well and long function names are not helpful when reading expressions.

I vote to rename hour_of_day() to hour() and month_of_year() to month(). The day_* functions make sense, even though I'd prefer day() over day_of_month().